src/couch_replicator_js_functions.hrl

```diff
@@ -172,8 +172,11 @@
 
 -define(REP_DB_TERMINAL_STATE_VIEW_MAP_FUN, <<"
     function(doc) {
-        if (typeof doc._replication_state === 'string') {
-            emit(doc._replication_state, doc._replication_state_reason);
+        state = doc._replication_state;
+        if (state === 'failed') {
+            emit('failed', doc._replication_state_reason);
+        } else if (state === 'completed') {
+            emit('completed', doc._replication_stats);
         }
```

Comment:
Does this need to emit for all states? You're changing the view to only emit when it's failed or completed.